### PR TITLE
Show 2 recently edited cohorts on dashboard page

### PR DESCRIPTION
**Pull request by eveadele**
https://app.asana.com/0/1199528021095056/1202199381153808/f

If the user has no cohorts at all, they will see an empty state prompting them to create a cohort. I added recentCohorts as a separate request with a separate key in the redux store.

Implements dashboard design. I'd love some feedback on the structure/organization of the code -- this is my first PR on this codebase so I took my best guess on where things belong. I'm also not sure what level of testing is necessary for this feature. There seem to be no tests at all for dashboard functionality -- is it worth a feature test? If not, which parts should be tested?

### CODE REVIEW

**Comment by enatario** [comment]
Hi! There is a design in the Figma that covers this scenario. I thought I had a static version of this little nugget in the code from my previous branch but maybe not?

Screen Shot 2022-05-11 at 9 15 11 AM

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "no-console": ["error", { "allow": ["warn", "error"] }],
+    "no-use-before-define": ["error"],
+    "react/display-name": ["off"]
+  },
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
+  }
+}
```
Comment (rwaldron) [inline .eslintrc:42]: This is certainly welcome, but I generally expect such changes to be done in dedicated pull requests. I also don't want to add extra work to your plate, so maybe we can just keep it here and any further "developer support" changes can be done in dedicated PRs.

File: client/components/Dashboard/RecentCohorts.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return (
+      <ul className="dashboard-grid">
+        {cohorts.map(cohort => {
+          return (
+            <li key={cohort.id}>
+              <CohortCard id={cohort.id} />
+            </li>
+          );
+        })}
+      </ul>
```
Comment (rwaldron) [inline client/components/Dashboard/RecentCohorts.jsx:23]: Please use `<List>` (https://react.semantic-ui.com/elements/list/) from `@components/UI`

File: client/components/Dashboard/RecentCohorts.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    );
+  } else {
+    return (
+      <div className="dashboard-empty">
+        <p>
+          No cohorts created. <a href="#">Create a new cohort.</a>
+        </p>
+      </div>
+    );
```
Comment (rwaldron) [inline client/components/Dashboard/RecentCohorts.jsx:32]: I suspect this whole else statement block is going to change, according to @enatario's feedback here: https://github.com/mit-teaching-systems-lab/dcss/pull/11#issuecomment-1123748896, but when implementing that, please be sure to use the appropriate semantic ui component. Thanks!

File: client/components/Dashboard/RecentCohorts.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        >
+          <Icon name="add" />
+          Create a new cohort
+        </Button>
+        <Button size="small" href="/">
+          View all cohorts
+        </Button>
+      </div>
+      <CohortContent cohorts={cohorts} />
+    </section>
```
Comment (rwaldron) [inline client/components/Dashboard/RecentCohorts.jsx:66]: Please the appropriate semantic ui components to construct this

**Review by rwaldron — CHANGES_REQUESTED** [review]
Back end looks great, front end logic looks great. Please address @enatario's design feedback and my notes regarding use of semantic ui components.

File: .eslintrc (hunk 1 above)
Comment (eveadele) [reply .eslintrc:42]: I didn't mean this to be a part of this PR! I'll revert it and put it in its own branch. Thanks!

**Comment by enatario** [comment]
@eveadele lgtm!

**Comment by rwaldron** [comment]
> Does this look okay for now?

Yes

> I had intended to handle the case where the user cannot create cohorts or scenarios in this task: https://app.asana.com/0/1202178937859057/1202199381153818. 

I don't think that addresses the issue. Most participants are never going to be facilitators and it won't make sense for a participant who is not in any cohorts to just see a suggestion that they should create a cohort, or should ask for permission to create cohorts.

**Comment by enatario** [comment]
> Most participants are never going to be facilitators and it won't make sense for a participant who is not in any cohorts to just see a suggestion that they should create a cohort, or should ask for permission to create cohorts.

That's a fair point. Are we saying that we should drop the dashboard altogether for participants since it's geared toward authors anyways? I suppose, would participants automatically see the dashboard if their way of accessing the site is primarily through a direct link to a cohort scenario.

Another question is how might a person _get_ to be an author. I'm trying to recall all the permission levels. If someone is coming in and making a new account and _wants_ to be an author, how would they do so through the site if not through that suggestion?

**Comment by rwaldron** [comment]
> Another question is how might a person get to be an author. I'm trying to recall all the permission levels. If someone is coming in and making a new account and wants to be an author, how would they do so through the site if not through that suggestion?

- If the user is only a **Participant**, or is  **Participant** that has **Researcher** privileges, show the two most recent cohorts the are currently in. Do not show the "create cohort" button. If they are in no cohorts, do not display anything cohort related. 
- If the user has **Facilitator**, **Admin** or **Super Admin** privileges, show the most recent two cohorts they created. If there are none, show the "Create a cohort button" (this is what is designed now)


As for requesting **Facilitator** privileges (which is the level that allows for authoring): presently, where the "Create a Cohort" and "Create a Scenario" buttons are displayed for current **Facilitator** users, a button could be displayed that says something like "I want to create Scenarios" or "I want to create Cohorts", and clicking that would create a requst for that user to be granted that privilege.

**Comment by enatario** [comment]
Cool that all makes sense to me! I think that's a good path forward as far as user experience. Do you need any design support for that?

**Review by rwaldron — APPROVED** [review]

**Pull request merged**